ifrs17-template/Initialization/InitSystemorphUtils.ipynb

```diff
@@ -0,0 +1,170 @@
+                "\n                                                                                        (\"TransactionalData/SimpleValue_CH_2020_12.csv\", ImportFormats.SimpleValue), ",
+                "\n                                                                                        (\"TransactionalData/NominalCashflows_CH_2020_12_MTUP10pct.csv\", ImportFormats.Cashflow), ",
+                "\n                                                                                        (\"TransactionalData/NominalCashflows_CH_2021_3.csv\", ImportFormats.Cashflow), ",
+                "\n                                                                                        (\"TransactionalData/Actuals_CH_2021_3.csv\", ImportFormats.Actual)}}};"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Import Tasks"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public async Task InitializeDataNodesAsync(IDataSourceVariable dataSource, string pathToImport, string[] nodes = null){",
+                "\n    nodes = nodes ?? new string[]{\"ES\", \"FR\", \"CH\"};",
+                "\n    var uploads = DataNodeFiles.Where(kvp => nodes.Contains(kvp.Key)).SelectMany(kvp => kvp.Value).ToArray();",
+                "\n    foreach(var pair in uploads) ",
+                "\n        await Import.FromFile(pathToImport + pair.Item1).WithFormat(pair.Item2).WithTarget(dataSource).WithActivityLog().ExecuteAsync();",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public async Task InitializeDataNodesAsync(IWorkspace workspace, string pathToImport, string[] nodes = null){",
+                "\n    nodes = nodes ?? new string[]{\"ES\", \"FR\", \"CH\"};",
+                "\n    var uploads = DataNodeFiles.Where(kvp => nodes.Contains(kvp.Key)).SelectMany(kvp => kvp.Value).ToArray();",
+                "\n    foreach(var pair in uploads)",
+                "\n        await Import.FromFile(pathToImport + pair.Item1).WithFormat(pair.Item2).WithTarget(workspace).WithActivityLog().ExecuteAsync();",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public async Task InitializeTransactionalDataAsync(IDataSourceVariable dataSource, string pathToImport, string[] nodes = null){",
```

Comment:
What type is IDataSourceVariable? Why not IDataSource?

Reply:
In principle both these interfaces are implemented by the DataSource. However your suggestion is better because IDataSource (unlike IDataSource Variable) is implemented both by Workspace and the DataSource, so your suggestion is more generic and compact than mine. I will change accordingly.